src/main/kotlin/no/nav/familie/tilbake/sikkerhet/AuditLogger.kt

```diff
@@ -0,0 +1,78 @@
+package no.nav.familie.tilbake.sikkerhet
+
+import no.nav.familie.log.mdc.MDCConstants
+import no.nav.familie.tilbake.common.ContextService
+import org.slf4j.LoggerFactory
+import org.slf4j.MDC
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.stereotype.Component
+import org.springframework.web.context.request.RequestContextHolder
+import org.springframework.web.context.request.ServletRequestAttributes
+import javax.servlet.http.HttpServletRequest
+
+/**
+ * [custom1], [custom2], [custom3] brukes for å logge ekstra felter, eks fagsak, behandling,
+ * disse logges til cs3,cs5,cs6 då cs1,cs2 og cs4 er til internt bruk
+ * Kan brukes med eks CustomKeyValue(key=fagsak, value=fagsakId)
+ */
+data class Sporingsdata(val event: AuditLoggerEvent,
+                        val personIdent: String,
+                        val custom1: CustomKeyValue? = null,
+                        val custom2: CustomKeyValue? = null,
+                        val custom3: CustomKeyValue? = null)
+
+enum class AuditLoggerEvent(val type: String) {
+    CREATE("create"),
+    UPDATE("update"),
+    ACCESS("access"),
+    NONE("Not logged")
+}
+
+data class CustomKeyValue(val key: String, val value: String)
+
+@Component
+class AuditLogger(@Value("\${NAIS_APP_NAME:appName}") private val applicationName: String) {
+
+    private val logger = LoggerFactory.getLogger(javaClass)
+    private val audit = LoggerFactory.getLogger("auditLogger")
+
+    fun log(data: Sporingsdata) {
+        val request = getRequest() ?: throw IllegalArgumentException("Ikke brukt i context av en HTTP request")
+
+        if (!ContextService.erMaskinTilMaskinToken()) {
+            audit.info(createAuditLogString(data, request))
+        } else {
+            logger.debug("Maskin til maskin token i request")
+        }
+    }
+
+    private fun getRequest(): HttpServletRequest? {
+        return RequestContextHolder.getRequestAttributes()
+                ?.takeIf { it is ServletRequestAttributes }
+                ?.let { it as ServletRequestAttributes }
+                ?.request
+    }
+
+    private fun createAuditLogString(data: Sporingsdata, request: HttpServletRequest): String {
+        val timestamp = System.currentTimeMillis()
+        val name = "Saksbehandling"
+        return "CEF:0|$applicationName|auditLog|1.0|audit:${data.event.type}|$name|INFO|end=$timestamp " +
+               "suid=${ContextService.hentSaksbehandler()} " +
+               "duid=${data.personIdent} " +
```

Comment:
Er det greit å logge personIdent i vanlig log? Kan vi bruke secureLogger her?

Reply:
Denne går til en egen audit-logg. Men siden jeg hadde glemt å legge til konfigurasjonen for audit-loggen, så skjønner jeg at du lurer.